src/main/java/it/gov/pagopa/reward/service/lookup/DeleteHpanServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package it.gov.pagopa.reward.service.lookup;
+
+import it.gov.pagopa.reward.dto.HpanInitiativeDTO;
+import it.gov.pagopa.reward.model.ActiveTimeInterval;
+import it.gov.pagopa.reward.model.HpanInitiatives;
+import it.gov.pagopa.reward.model.OnboardedInitiative;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalTime;
+import java.util.Comparator;
+import java.util.List;
+
+@Service
+@Slf4j
+public class DeleteHpanServiceImpl implements DeleteHpanService{
+    @Override
+    public HpanInitiatives execute(HpanInitiatives hpanInitiatives, HpanInitiativeDTO hpanInitiativeDTO) {
+        if (hpanInitiatives.getHpan() != null) {
+            List<OnboardedInitiative> onboardedInitiatives = hpanInitiatives.getOnboardedInitiatives();
+            log.trace("[DELETED_HPAN] Is presente in db the follow hpan: {}",hpanInitiativeDTO.getHpan());
+            if (onboardedInitiatives != null){
+               OnboardedInitiative onboardedInitiative = onboardedInitiatives.stream().filter(o -> o.getInitiativeId().equals(hpanInitiativeDTO.getInitiativeId())).findFirst().orElse(null);
+                if (onboardedInitiative!=null) {
+                    return evaluateHpanWithInitiativePresent(hpanInitiatives, hpanInitiativeDTO, onboardedInitiative);
+                }
+                log.debug("The hpan has no reference to the initiative. Source message: {}", hpanInitiativeDTO);
+                return null;
+            }
+            log.error("Unexpected use case, the hpan not have any initiatives associate. Source message: {}", hpanInitiativeDTO);
+            return null;
+        }
+        log.debug("The hpan is not present into DB. Source message: {}", hpanInitiativeDTO);
+        return null;
```

Comment:
put these line into else branch, this way we are sure of the use case (if changing the code in the future we add some other conditional code that could brought to go out from the previous statement, it will be covered by this return!

Moreover, log it as ERROR level